airbyte-webapp/src/views/Connector/ServiceForm/components/AuthButtonFactory.tsx

```diff
@@ -0,0 +1,58 @@
+import { Button } from "../../../../components/base/Button";
+import { FormattedMessage } from "react-intl";
+import { ConnectorDefinition } from "../../../../core/domain/connector";
+import { isSourceDefinition } from "../../../../core/domain/connector/source";
+import { isDestinationDefinition } from "../../../../core/domain/connector/destination";
+
+const getButton = (
+  loading: boolean,
+  run: () => void,
+  selectedService?: ConnectorDefinition,
+  done?: boolean
+): JSX.Element => {
+  let definitionId = "";
+  if (selectedService && isSourceDefinition(selectedService)) {
+    definitionId = selectedService.sourceDefinitionId;
+  } else if (selectedService && isDestinationDefinition(selectedService)) {
+    definitionId = selectedService.destinationDefinitionId;
+  } else {
+    throw new Error(`Unrecognized service type. Service: ${selectedService}`);
```

Comment:
Very bad idea to throw an error. ConnectorDefinition is always either SourceDefinition or DestinationDefinition: how is it possible to throw an error then?

Reply:
It generally should not happen. This is to future proof the case where let's say we add a new type of definition and it is not handled here for whatever reason. I can remove the error though if that's not a pattern followed here

Reply:
I think in such cases we should better display users something like an unexpected error happened or something like that. Because in case error is thrown this way - the app could fail ( or user will be redirected back to the last working state ). So instead of handling unexpected case - we will fail silently when an error is thrown and will have to somehow handle such case in `<ErrorBoundary>`